File: src/main/kotlin/p26xx/Problem2639.kt
package p26xx

import util.expect

fun main() {
    class Solution {
        fun findColumnWidth(grid: Array<IntArray>): IntArray {
            val result = IntArray(grid.map { it.size }.max())

            grid.forEach {
                it.forEachIndexed { index, i ->
                    result[index] = result[index].coerceAtLeast(i.toString().length)
                }
            }

            return result
        }
    }

    expect {
        Solution().findColumnWidth(
            arrayOf()
        )
    }
}

File: src/main/kotlin/p26xx/Problem2639.ts
function findColumnWidth(grid: number[][]): number[] {
    const result : number[] = []

    for (const row of grid){
        for (let i = 0; i < row.length; i++){
            result[i] = Math.max(result[i]??0, row[i].toString().length)
        }
    }

    return result
};